atlasdb-client/src/main/java/com/palantir/atlasdb/AtlasDbConstants.java

```diff
@@ -80,6 +80,7 @@ private AtlasDbConstants() {
     public static final long MAX_TS = Long.MAX_VALUE;
 
     public static final long DEFAULT_TRANSACTION_LOCK_ACQUIRE_TIMEOUT_MS = 60_000;
+    public static final int THRESHOLD_FOR_LOGGING_LARGE_NUMBER_OF_TRANSACTION_LOOKUPS = 10_000;
```

Comment:
Is this the right threshold?  I could set it ridiculously high so users have to 'enable' it by setting a sensible value in config.

Reply:
Added a Histogram

Reply:
it also may be interesting as a follow up for this to have the theshold be configurable/overridable per table.